### PR TITLE
[VL] Customize VCPKG build features according to user's build options

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

When vcpkg is enabled, installing native dependencies related to test, hdfs, s3, gcs and abfs should be optional, which depends on user-specified build options. We can put always-required dependencies as vcpkg default feature and enable other feature according to build option. 

## How was this patch tested?
CI.

### CODE REVIEW

**Review by rui-mo — APPROVED** [review]

File: dev/vcpkg/env.sh [hunk 1]
```diff
@@ -7,5 +7,25 @@ if [ -z "${BASH_SOURCE[0]}" ] || [ "$0" == "${BASH_SOURCE[0]}" ]; then
...
+init_vcpkg_env=$("${SCRIPT_ROOT}/init.sh" $@)
 eval "$init_vcpkg_env"
+
+export VCPKG_ROOT="$SCRIPT_ROOT/.vcpkg"
+export VCPKG="$SCRIPT_ROOT/.vcpkg/vcpkg"
+export VCPKG_TRIPLET=x64-linux-avx
+export VCPKG_TRIPLET_INSTALL_DIR=${SCRIPT_ROOT}/vcpkg_installed/${VCPKG_TRIPLET}
+export EXPORT_TOOLS_PATH="${VCPKG_TRIPLET_INSTALL_DIR}/tools/protobuf"
+
+if [ "\${GLUTEN_VCPKG_ENABLED:-}" != "${VCPKG_ROOT}" ]; then
```
Comment (zhztheplayer) [inline dev/vcpkg/env.sh:19]: `"\${GLUTEN_VCPKG_ENABLED:-}"` or `"${GLUTEN_VCPKG_ENABLED:-}"`?

File: dev/vcpkg/env.sh [hunk 2]
```diff
@@ -7,5 +7,25 @@ if [ -z "${BASH_SOURCE[0]}" ] || [ "$0" == "${BASH_SOURCE[0]}" ]; then
...
+    export VCPKG_ROOT=${VCPKG_ROOT}
+    export VCPKG_MANIFEST_DIR=${SCRIPT_ROOT}
+    export VCPKG_TRIPLET=${VCPKG_TRIPLET}
+
+    export CMAKE_TOOLCHAIN_FILE=${SCRIPT_ROOT}/toolchain.cmake
+    export PKG_CONFIG_PATH=${VCPKG_TRIPLET_INSTALL_DIR}/lib/pkgconfig:${VCPKG_TRIPLET_INSTALL_DIR}/share/pkgconfig:\${PKG_CONFIG_PATH:-}
```
Comment (zhztheplayer) [inline dev/vcpkg/env.sh:25]: `\${PKG_CONFIG_PATH:-}` vs `${PKG_CONFIG_PATH:-}`?

File: dev/vcpkg/env.sh [hunk 3]
```diff
@@ -7,5 +7,25 @@ if [ -z "${BASH_SOURCE[0]}" ] || [ "$0" == "${BASH_SOURCE[0]}" ]; then
 fi
 
 SCRIPT_ROOT="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
-init_vcpkg_env=$("${SCRIPT_ROOT}/init.sh")
+init_vcpkg_env=$("${SCRIPT_ROOT}/init.sh" $@)
 eval "$init_vcpkg_env"
```
Comment (zhztheplayer) [inline dev/vcpkg/env.sh:11]: Do we still need this `eval`?

**Comment by zhztheplayer** [comment]
@PHILO-HE 

The change breaks one of my daily benchmark jobs. Relevant log:

https://gist.github.com/zhztheplayer/f3f4d946d42060694c743f32ab38ba7b

**Comment by zhztheplayer** [comment]
@PHILO-HE I am testing a fix https://github.com/apache/incubator-gluten/pull/7315

**Pull request merged**